docs/source/index.rst

```diff
@@ -2,17 +2,38 @@
 opt_einsum
 ==========
 
-:func:`~numpy.einsum` is a powerful function for contracting tensors of arbitrary
-dimension and index. However, it is only optimized to contract two terms
-at a time resulting in non-optimal scaling.
+Optimized einsum can greatly reduce the overall execution time of einsum-like
+expressions by optimizing the expression's contraction order and dispatching
+many operations to canonical BLAS, cuBLAS, or other specialized routines.
+Optimized einsum is agnostic to the backend and can handle NumPy, Dask,
+PyTorch, Tensorflow, CuPy, Sparse, Theano, Jax, and Autograd arrays.
```

Comment:
Same comment about potentially arbitrary other backends as above! But I'll leave to you @dgasmith to decide if its worth emphasizing this here as well.